.github/workflows/ensure-blocking-pr-labels-absent.yml

```diff
@@ -13,13 +13,19 @@ jobs:
     permissions:
       pull-requests: read
     steps:
-      - uses: actions/checkout@v3
-      - name: Set up Node.js
-        uses: actions/setup-node@v3
+      - uses: actions/checkout@v4
+      - name: Use Node.js
+        uses: actions/setup-node@v4
```

Comment:
Strange that we're calling `setup-node` here twice

Reply:
It is definitely strange, yes, but there is a reason for it. We need to call it once to be able to use Corepack to install Yarn (recent versions of Node ship with Corepack). However, we can't restore the Yarn cache in this step because we need the correct version of Yarn to do that, and we don't have it yet. So after we install Node once without restoring the Yarn cache, we then install Corepack. Then we run the `setup-node` action again, this time restoring the Yarn cache.

The alternative for this would be to ship the Yarn binary with the repo as we did before instead of using Corepack. Lego seemed to not like this way because then we need to guarantee that the Yarn binary is correct everywhere (which we never did): https://github.com/MetaMask/metamask-module-template/pull/243#issuecomment-2083838008. But we could open up that discussion again.

See relevant module template PRs:

- https://github.com/MetaMask/metamask-module-template/pull/243
- https://github.com/MetaMask/metamask-module-template/pull/244
- https://github.com/MetaMask/metamask-module-template/pull/246